src/helpers.c

```diff
@@ -135,8 +135,8 @@ static int create_required_dirs(void)
 	int ret = 0;
 	int i;
 	char *dir;
-#define STATE_DIR_COUNT 3
-	const char *dirs[] = { "delta", "staged", "download" };
+#define STATE_DIR_COUNT 4
+	const char *dirs[] = { "delta", "staged", "download", "telemetry" };
```

Comment:
That would be an improvement, though it could be incorporated in a separate cleanup patch.